File: 02_Object_Oriented_Programming/03_Constructor_and_It's_Types/05_Interview_Questions/01_Question_2/python/main.py
class Example:
    def __init__(self, a, b):
        self.a = a
        self.b = b


try:
    example = Example()  # TypeError: missing required arguments
    print(example)
except TypeError as e:
    print(f"Error: {e}")

"""
In Java, this would cause a Compilation Error because
no default constructor exists when a parameterized constructor is defined.

In Python, this raises a TypeError at runtime.

Output:
Error: Example.__init__() missing 2 required positional arguments: 'a' and 'b'
"""

File: 02_Object_Oriented_Programming/03_Constructor_and_It's_Types/05_Interview_Questions/01_Question_2/javascript/main.js
class Example {
    constructor(a, b) {
        this.a = a;
        this.b = b;
    }
}

const example = new Example(); // No error in JS, but a and b will be undefined
console.log(example);

/*
In Java, this would cause a Compilation Error because
no default constructor exists when a parameterized constructor is defined.

In JavaScript, there is no compilation error, but a and b will be undefined.

Output:
Example { a: undefined, b: undefined }
*/

